Document properties of DataProcessors (#393)
TYPO3-Documentation/TYPO3CMS-Reference-Typoscript: Documentation/ContentObjects/Fluidtemplate/DataProcessing/MenuProcessor.rst

.. include:: /Includes.txt
.. _MenuProcessor:

=============
MenuProcessor
=============

The :php:`MenuProcessor` utilizes :ts:`HMENU` to generate a JSON encoded menu string

> that will be decoded again and assigned to :ts:`FLUIDTEMPLATE` as variable.

Reviewer: ```suggestion
that will be decoded again and assigned to :ts:`FLUIDTEMPLATE` as a variable.
```

Additional data processing is supported and will be applied to each record.

Options:
========

.. confval:: levels

   :Required: true
   :type: int, :ref:`stdWrap`
   :default: 1
   :Example: 5

   Maximal number of levels to be included in the output array

.. confval:: expandAll

   :Required: true
   :type: int, :ref:`stdWrap`
   :default: 1
   :Example: 0

   Should all submenus be included or only those of the active pages?

.. confval:: includeSpacer

   :Required: true
   :type: int, :ref:`stdWrap`
   :default: 0
   :Example: 1

   Include pages with type "spacer"

.. confval:: titleField

   :Required: true
   :type: int, :ref:`stdWrap`
   :default: "nav_title // title"
   :Example: "subtitle"

   Fields to be used as title

.. confval:: as

   :Required: false
   :type: string
   :default: "menu"

>    Name the variable in the Fluid template will have

Reviewer: ```suggestion
   Name for the variable in the Fluid template.
```
